genesis/camino_genesis_test.go

```diff
@@ -0,0 +1,290 @@
+// Copyright (C) 2022, Chain4Travel AG. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package genesis
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/ava-labs/avalanchego/database/memdb"
+	"github.com/ava-labs/avalanchego/database/versiondb"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/snow"
+	"github.com/ava-labs/avalanchego/utils/constants"
+	"github.com/ava-labs/avalanchego/utils/formatting/address"
+	"github.com/ava-labs/avalanchego/utils/math"
+	"github.com/ava-labs/avalanchego/utils/set"
+	"github.com/ava-labs/avalanchego/utils/units"
+	"github.com/ava-labs/avalanchego/utils/wrappers"
+	"github.com/ava-labs/avalanchego/vms/components/avax"
+	"github.com/ava-labs/avalanchego/vms/platformvm/deposit"
+	"github.com/ava-labs/avalanchego/vms/platformvm/genesis"
+	"github.com/ava-labs/avalanchego/vms/platformvm/reward"
+	"github.com/ava-labs/avalanchego/vms/secp256k1fx"
+	"github.com/stretchr/testify/require"
+
+	_ "embed"
+)
+
+var (
+	defaultRewardConfig = reward.Config{
+		MaxConsumptionRate: .12 * reward.PercentDenominator,
+		MinConsumptionRate: .10 * reward.PercentDenominator,
+		MintingPeriod:      365 * 24 * time.Hour,
+		SupplyCap:          720 * units.MegaAvax,
+	}
+	initialAdmin     = "X-kopernikus1m4nr983lhd4p4nfsqk3a6a9mejugnn73cmp283"
+	expectedUtxoID1  = wrappers.IgnoreError(ids.FromString("2P2kTVUNMvzVWAgTbaML2DRqpD9pj5gWmxGiKZGDbotqBCHdqJ")).(ids.ID)
+	expectedUtxoID2  = wrappers.IgnoreError(ids.FromString("m3qqtnFG78HozeZ2kSS3pTLeo2SKxjTiR3DJvBd1eKyQTDcnZ")).(ids.ID)
+	expectedBondTxID = wrappers.IgnoreError(ids.FromString("2iMQED4CRfaUZcUgseM7vHXA7fcuwDvjYjq8BfJPTyGb84H8HN")).(ids.ID)
+)
+
+func TestValidateCaminoConfig(t *testing.T) {
+	type args struct {
+		config *Config
+	}
+	tests := map[string]struct {
+		args args
+		err  error
+	}{
+		"non-camino allocations / LockModeBondDeposit=false": {
+			args: args{
+				config: func() *Config {
+					thisConfig := KopernikusConfig
+					thisConfig.Camino.LockModeBondDeposit = false
+					thisConfig.Allocations = []Allocation{{AVAXAddr: ids.GenerateTestShortID()}}
+					return &thisConfig
+				}(),
+			},
+		},
+		"non-camino allocations": {
+			args: args{
+				config: func() *Config {
+					thisConfig := KopernikusConfig
+					thisConfig.Allocations = []Allocation{{AVAXAddr: ids.GenerateTestShortID()}}
+					return &thisConfig
+				}(),
+			},
+			err: errors.New("config.Allocations != 0"),
+		},
+		"invalid deposit offer / start date after before date": {
+			args: args{
+				config: func() *Config {
+					thisConfig := KopernikusConfig
+					offer := KopernikusConfig.Camino.DepositOffers[0]
+					offer.Start = 2
+					offer.End = 1
+					offers := []genesis.DepositOffer{offer}
+					thisConfig.Camino.DepositOffers = offers
+					return &thisConfig
+				}(),
+			},
+			err: fmt.Errorf(
```

Comment:
plz, if u have time, make error vars with sane message, wrap them with this data that currently in fmt.Errorf, and then test for getting this wrapped errors
sorry for not making good errors in a first place :(

Reply:
done